### ISSUE TITLE
textDocument/codeAction fails without any further details.

### ISSUE THREAD

**Issue by apollo13** [microsoft/pyright#6981]
**Describe the bug**

I have checked out the ansible repo and when trying to hover certain epxressions I get this error:

```
[Error - 1:04:37 PM] Request textDocument/codeAction failed.
  Message: Request textDocument/codeAction failed unexpectedly without providing any details.
  Code: -32603 
```

I cannot figure out any details on why this occurs, sorry. Nevertheless it seems to be 100% reproducable on my machine when I hover `params` in https://github.com/ansible/ansible/blob/bf9509bf33b2675d73f4abe72bebb91815d284f5/lib/ansible/module_utils/basic.py#L386

**VS Code extension or command-line**
PyRight VSCode extension 1.1.345

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported.

A few follow-up questions:

Are you using the VS Code extension with the Microsoft official release of VS Code? If so, I recommend switching to Pylance, which is built on pyright but has a bunch of extra language server functionality.

Are you running any other Python language servers? In your settings, do you have `python.languageServer` set to "None", or is it set to "Default"? If it's the latter, then you're probably also running Jedi.

Which version of Python are you using?

Where does the error message appear? Is it a popup dialog in VS Code? Is it in the log "Output" for Pyright?

The error message references a "codeAction", but hovering shouldn't trigger any code actions, at least not in Pyright. Are you sure you don't have some other extensions installed?

**Comment by apollo13** [comment]
Hi @erictraut,

> I'm not able to repro the problem as reported.

Damn :/

> Are you using the VS Code extension with the Microsoft official release of VS Code? If so, I recommend switching to Pylance, which is built on pyright but has a bunch of extra language server functionality.

Yes, and no I rather not lock myself into that eco system.

> Are you running any other Python language servers? In your settings, do you have `python.languageServer` set to "None", or is it set to "Default"? If it's the latter, then you're probably also running Jedi.

Nope, this is the complete settings file:
```
{
    "workbench.startupEditor": "none",
    "python.languageServer": "None",
    "python.analysis.logLevel": "Trace",
}
```

> Which version of Python are you using?

3.11.7  -- which is kinda weird because `/bin/python` is `3.12` by now. Does vscode cache that somewhere?
![image](https://github.com/microsoft/pyright/assets/81547/9b3d67f7-cf55-45e0-870b-264e0b77b4d5)

> Where does the error message appear? Is it a popup dialog in VS Code? Is it in the log "Output" for Pyright?

Both:
![image](https://github.com/microsoft/pyright/assets/81547/8b65ba4b-5488-4eeb-a118-589e47ccfa73)
![image](https://github.com/microsoft/pyright/assets/81547/114b750b-de1b-4711-997d-bc33af368983)

> The error message references a "codeAction", but hovering shouldn't trigger any code actions, at least not in Pyright. Are you sure you don't have some other extensions installed?

Well no, the Python extension is also installed, but PyRight doesn't work without it iirc?
![image](https://github.com/microsoft/pyright/assets/81547/6b88b405-6ea9-4231-8c0e-49916c12f0b8)

**Comment by erictraut** [comment]
> the Python extension is also installed, but PyRight doesn't work without it iirc?

Pyright will run fine without the Python extension. However, the Python extension provides the convenient "Python environment picker" functionality. If you don't have the Python extension installed, then you'll need to manually specify your desired Python environment using the `python.pythonPath` setting.

Without a repro case, I won't be able to do anything with this bug report.

Which platform are you running on? I wouldn't think that would matter, but I figured I'd ask.

Did this just start happening? If you downgrade to a previous version of Pyright, does the problem go away?

I wonder if the VS Code extension is corrupt. We've seen that happen sometimes. You could try uninstalling it (and deleting from VS Code's cache), then reinstalling it.

Perhaps the Pylance team has some thoughts. @debonte, @rchiodo?

**Comment by apollo13** [comment]
> Without a repro case, I won't be able to do anything with this bug report.

I understand, but I literally just cloned it and hovered over the `params` in the linked line. I just disabled the Python extension and the issue still occurs. After setting `python.pythonPath` it shows:
```
[Info  - 6:18:46 PM] Setting pythonPath for service "ansible": "/bin/python"
[Info  - 6:18:46 PM] Loading pyproject.toml file at /home/florian/sources/ansible/ansible/pyproject.toml
[Info  - 6:18:46 PM] Pyproject file "/home/florian/sources/ansible/ansible/pyproject.toml" has no "[tool.pyright]" section.
[Info  - 6:18:46 PM] Assuming Python version 3.11
Assuming Python platform Linux
```
Any idea why it assumes 3.11? does it not execute `python --version` or anything?

> Which platform are you running on? I wouldn't think that would matter, but I figured I'd ask.

Fedora 39

> Did this just start happening? If you downgrade to a previous version of Pyright, does the problem go away?

Not sure, I don't use that code often. I just bisected and it seems that the issue occurs since 1.1.340. The earliest version I tested was 1.1.299 and then I worked myself up in a binary search fashion to 1.1.340 -- maybe that release didn't have to many commits?

**Comment by erictraut** [comment]
To confirm, you're not able to repro the issue in 1.1.339 but you are with 1.1.340?

**Comment by apollo13** [comment]
Yes exactly.

**Comment by apollo13** [comment]
Aha, getting somewhere, with 340 the line looks like this:
![image](https://github.com/microsoft/pyright/assets/81547/23b53968-c3f8-4a3e-b91d-eec4af3638bc)
and with 339:
![image](https://github.com/microsoft/pyright/assets/81547/09c4f6bd-239e-490a-a9c2-99448e918238)

So apparently due to the problem detection there is a code action?

**Comment by erictraut** [comment]
Thanks for the additional clues. I was able to find the repro case. The reason I wasn't seeing it before is that 1) it occurs only in production builds, not in the debug build that I normally use during development, and 2) it has already been fixed in the main branch by a commit that came in via the pylance team.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
